File: TS/系统/03．技能系统/00．技能模板+函数/01．技能函数/20．物品辅助/14．数值伤害与治疗.ts
/** @noSelfInFile */

import { 单位存活 } from "./12．物品与单位";
import { 造成装备技能伤害 } from "../../../../04．伤害系统/08．技能伤害系统";

const jass = require("jass.common") as any;
const japi = require("jass.japi") as any;

const { UnitHasItemOfTypeBJ, GetItemOfTypeFromUnitBJ } = require("lib.扩展函数.物品相关函数.物品判断函数") as {
  UnitHasItemOfTypeBJ: (this: void, whichUnit: any, itemTypeId: number) => boolean;
  GetItemOfTypeFromUnitBJ: (this: void, whichUnit: any, itemId: number) => any | null;
};
const { getUnitsInRange, getEnemyUnitsInRange } = require("lib.扩展函数.自定义扩展函数.01．选取中心范围") as {
  getUnitsInRange: (this: void, x: number, y: number, radius: number) => any[];
  getEnemyUnitsInRange: (this: void, centerUnit: any, x: number, y: number, radius: number) => any[];
};
const { SGSS_SetState } = require("lib.扩展函数.Star扩展函数.00．SGSS") as {
  SGSS_SetState: (this: void, unit: any, id: number, value: number) => void;
};
const { YDUserDataGetSafe, YDUserDataSetSafe } = require("lib.扩展函数.YDWE函数.09．YDUserData安全版") as {
  YDUserDataGetSafe: (this: void, tableType: string, tableKey: any, attr: string, valueType: string) => any;
  YDUserDataSetSafe: (this: void, tableType: string, tableKey: any, attr: string, valueType: string, value: any) => void;
};
const { doHeal } = require("系统.04．伤害系统.02．治疗系统.01．核心功能") as {
  doHeal: (this: void, params: any) => number;
};
const { SFB_setBuff, SFB_setSlow } = require("系统.03．技能系统.00．技能模板+函数.02．通用函数.01．控制与Buff") as {
  SFB_setBuff: (this: void, source: any, target: any, id: number, time: number) => void;
  SFB_setSlow: (this: void, source: any, target: any, attackSlow: number, moveSlow: number, time: number) => void;
};
const { 清除单位负面Buff } = require("系统.03．技能系统.00．技能模板+函数.02．通用函数.01．控制与Buff") as {
  清除单位负面Buff: (this: void, unit: any, onlyPurgable?: boolean) => number;
};
const { 开始击退 } = require("系统.03．技能系统.00．技能模板+函数.01．技能函数.02．冲锋·击退.击退系统") as {
  开始击退: (this: void, unit: any, params: any) => number;
};
const { addPeriodicCallback, getServerTime } = require("系统.00．核心系统.05．中心计时器") as {
  addPeriodicCallback: (this: void, intervalMs: number, callback: () => void) => number;
  getServerTime: (this: void) => number;
};
const { 创建单位并登记排泄安全 } = require("lib.扩展函数.自定义扩展函数.05．单位相关安全包装") as {
  创建单位并登记排泄安全: (this: void, owner: any, unitTypeId: number, x: number, y: number, facing: number) => any;
};

const GetItemTypeId = jass.GetItemTypeId as (whichItem: any) => number;
const GetHandleId = jass.GetHandleId as (h: any) => number;
const GetOwningPlayer = jass.GetOwningPlayer as (whichUnit: any) => any;
const GetPlayerId = jass.GetPlayerId as (whichPlayer: any) => number;
const GetUnitX = jass.GetUnitX as (whichUnit: any) => number;
const GetUnitY = jass.GetUnitY as (whichUnit: any) => number;
const GetUnitState = jass.GetUnitState as (whichUnit: any, whichUnitState: any) => number;
const SetUnitState = jass.SetUnitState as (whichUnit: any, whichUnitState: any, newVal: number) => void;
const IsUnitType = jass.IsUnitType as (whichUnit: any, whichUnitType: any) => boolean;
const IsUnitAlly = jass.IsUnitAlly as (whichUnit: any, whichPlayer: any) => boolean;
const GetHeroStr = jass.GetHeroStr as (whichHero: any, includeBonuses: boolean) => number;
const GetHeroAgi = jass.GetHeroAgi as (whichHero: any, includeBonuses: boolean) => number;
const GetHeroInt = jass.GetHeroInt as (whichHero: any, includeBonuses: boolean) => number;
const AddHeroXP = jass.AddHeroXP as (whichHero: any, xpToAdd: number, showEyeCandy: boolean) => void;
const ModifyHeroStat = jass.ModifyHeroStat as (whichStat: any, whichHero: any, modifyMethod: any, value: number) => void;
const AddSpecialEffect = jass.AddSpecialEffect as (modelName: string, x: number, y: number) => any;
const AddSpecialEffectTarget = jass.AddSpecialEffectTarget as (modelName: string, targetWidget: any, attachPointName: string) => any;
const DestroyEffect = jass.DestroyEffect as (whichEffect: any) => boolean;
const IsPointBlighted = jass.IsPointBlighted as (x: number, y: number) => boolean;
const SetItemCharges = jass.SetItemCharges as (whichItem: any, charges: number) => void;
const GetItemCharges = jass.GetItemCharges as (whichItem: any) => number;
const UnitApplyTimedLife = jass.UnitApplyTimedLife as (whichUnit: any, buffId: number, duration: number) => void;
const SetUnitScale = jass.SetUnitScale as (whichUnit: any, scaleX: number, scaleY: number, scaleZ: number) => void;
const SetUnitInvulnerable = jass.SetUnitInvulnerable as (whichUnit: any, flag: boolean) => void;
const SetUnitFacing = jass.SetUnitFacing as (whichUnit: any, facingAngle: number) => void;
const IssueTargetOrder = jass.IssueTargetOrder as (whichUnit: any, order: string, targetWidget: any) => boolean;
const CreateGroup = jass.CreateGroup as () => any;
const GroupEnumUnitsInRange = jass.GroupEnumUnitsInRange as (whichGroup: any, x: number, y: number, radius: number, filter: any) => void;
const FirstOfGroup = jass.FirstOfGroup as (whichGroup: any) => any;
const GroupRemoveUnit = jass.GroupRemoveUnit as (whichGroup: any, whichUnit: any) => void;
const DestroyGroup = jass.DestroyGroup as (whichGroup: any) => void;
const GetUnitFlyHeight = jass.GetUnitFlyHeight as (whichUnit: any) => number;
const ConvertUnitState = jass.ConvertUnitState as (unitStateId: number) => any;
const SquareRoot = jass.SquareRoot as (x: number) => number;
const Atan2 = jass.Atan2 as (y: number, x: number) => number;
const Cos = jass.Cos as (radians: number) => number;
const Sin = jass.Sin as (radians: number) => number;
const bj_RADTODEG = jass.bj_RADTODEG as number;
const bj_DEGTORAD = jass.bj_DEGTORAD as number;
const UNIT_TYPE_HERO = jass.UNIT_TYPE_HERO as any;
const UNIT_TYPE_DEAD = jass.UNIT_TYPE_DEAD as any;
const UNIT_TYPE_MECHANICAL = jass.UNIT_TYPE_MECHANICAL as any;
const UNIT_TYPE_ANCIENT = jass.UNIT_TYPE_ANCIENT as any;
const UNIT_STATE_LIFE = jass.UNIT_STATE_LIFE as any;
const UNIT_STATE_MANA = jass.UNIT_STATE_MANA as any;
const UNIT_STATE_MAX_LIFE = jass.UNIT_STATE_MAX_LIFE as any;
const UNIT_STATE_MAX_MANA = jass.UNIT_STATE_MAX_MANA as any;
const ATTACK_TYPE_NORMAL = jass.ATTACK_TYPE_NORMAL as any;
const DAMAGE_TYPE_NORMAL = jass.DAMAGE_TYPE_NORMAL as any;
const DAMAGE_TYPE_ENHANCED = jass.DAMAGE_TYPE_ENHANCED as any;
const DAMAGE_TYPE_FIRE = jass.DAMAGE_TYPE_FIRE as any;
const DAMAGE_TYPE_SHADOW_STRIKE = jass.DAMAGE_TYPE_SHADOW_STRIKE as any;
const DAMAGE_TYPE_MIND = jass.DAMAGE_TYPE_MIND as any;
const WEAPON_TYPE_WHOKNOWS = jass.WEAPON_TYPE_WHOKNOWS as any;
const bj_HEROSTAT_INT = jass.bj_HEROSTAT_INT as any;
const bj_MODIFYMETHOD_ADD = jass.bj_MODIFYMETHOD_ADD as any;
const GetUnitStateJapi = japi.GetUnitState as (whichUnit: any, whichUnitState: any) => number;
const DzSetUnitModel = japi.DzSetUnitModel as (whichUnit: any, model: string) => void;

const stringToFourCCSafe = (require("lib.扩展函数.封装函数.01．通用工具.01．FourCC转换安全版") as {
  stringToFourCCSafe: (this: void, s: string | undefined | null) => number;
}).stringToFourCCSafe;

const 火把单位类型ID = stringToFourCCSafe("e00D");
const 限时生命BuffID = stringToFourCCSafe("BHwe");

type 待销毁特效记录 = {
  句柄: any;
  到期时间: number;
};

const 待销毁特效列表: 待销毁特效记录[] = [];
let 已注册特效销毁驱动 = false;

function 处理待销毁特效(): void {
  const 当前时间 = getServerTime();
  for (let i = 待销毁特效列表.length - 1; i >= 0; i--) {
    const 记录 = 待销毁特效列表[i];
    if (当前时间 < 记录.到期时间) continue;
    DestroyEffect(记录.句柄);
    待销毁特效列表.splice(i, 1);
  }
}

function 安排特效销毁(effect: any, 持续秒: number = 1): void {
  if (effect == null || effect === 0) return;
  if (!已注册特效销毁驱动) {
    已注册特效销毁驱动 = true;
    addPeriodicCallback(100, 处理待销毁特效);
  }
  待销毁特效列表.push({
    句柄: effect,
    到期时间: getServerTime() + 持续秒 * 1000,
  });
}

export function 取当前生命(this: void, 单位: any): number {
  return GetUnitState(单位, UNIT_STATE_LIFE);
}

export function 取当前魔法(this: void, 单位: any): number {
  return GetUnitState(单位, UNIT_STATE_MANA);
}

export function 取最大生命(this: void, 单位: any): number {
  return GetUnitStateJapi(单位, UNIT_STATE_MAX_LIFE);
}

export function 取最大魔法(this: void, 单位: any): number {
  return GetUnitStateJapi(单位, UNIT_STATE_MAX_MANA);
}

export function 取单位攻击(this: void, 单位: any): number {
  return GetUnitStateJapi(单位, ConvertUnitState(0x15));
}

export function 设置生命(this: void, 单位: any, 数值: number): void {
  SetUnitState(单位, UNIT_STATE_LIFE, 数值);
}

export function 设置魔法(this: void, 单位: any, 数值: number): void {
  SetUnitState(单位, UNIT_STATE_MANA, 数值);
}

export function 调整生命(this: void, 单位: any, 数值: number): void {
  SetUnitState(单位, UNIT_STATE_LIFE, GetUnitState(单位, UNIT_STATE_LIFE) + 数值);
}

export function 调整魔法(this: void, 单位: any, 数值: number): void {
  SetUnitState(单位, UNIT_STATE_MANA, GetUnitState(单位, UNIT_STATE_MANA) + 数值);
}

export function 造成强化伤害(this: void, 来源: any, 目标: any, 伤害: number): void {
  if (!单位存活(来源) || !单位存活(目标) || !(伤害 > 0)) return;
  造成装备技能伤害({ 来源, 目标, 伤害, 伤害类型: DAMAGE_TYPE_ENHANCED, ranged: false, attackType: ATTACK_TYPE_NORMAL, weaponType: WEAPON_TYPE_WHOKNOWS });
}

export function 造成火焰伤害(this: void, 来源: any, 目标: any, 伤害: number): void {
  if (!单位存活(来源) || !单位存活(目标) || !(伤害 > 0)) return;
  造成装备技能伤害({ 来源, 目标, 伤害, 伤害类型: DAMAGE_TYPE_FIRE, ranged: true, attackType: ATTACK_TYPE_NORMAL, weaponType: WEAPON_TYPE_WHOKNOWS });
}

export function 造成暗影伤害(this: void, 来源: any, 目标: any, 伤害: number): void {
  if (!单位存活(来源) || !单位存活(目标) || !(伤害 > 0)) return;
  造成装备技能伤害({ 来源, 目标, 伤害, 伤害类型: DAMAGE_TYPE_SHADOW_STRIKE, ranged: true, attackType: ATTACK_TYPE_NORMAL, weaponType: WEAPON_TYPE_WHOKNOWS });
}

export function 造成普通伤害(this: void, 来源: any, 目标: any, 伤害: number): void {
  if (!单位存活(来源) || !单位存活(目标) || !(伤害 > 0)) return;
  造成装备技能伤害({ 来源, 目标, 伤害, 伤害类型: DAMAGE_TYPE_NORMAL, ranged: false, attackType: ATTACK_TYPE_NORMAL, weaponType: WEAPON_TYPE_WHOKNOWS });
}

export function 造成精神自伤(this: void, 单位: any, 伤害: number): void {
  if (!单位存活(单位) || !(伤害 > 0)) return;
  造成装备技能伤害({ 来源: 单位, 目标: 单位, 伤害, 伤害类型: DAMAGE_TYPE_MIND, ranged: false, attackType: ATTACK_TYPE_NORMAL, weaponType: WEAPON_TYPE_WHOKNOWS });
}

export function 执行治疗(this: void, 来源: any, 目标: any, 生命: number, 魔法: number = 0): void {
  if (目标 == null || 目标 === 0) return;
  doHeal({
    HealSource: 来源,
    HealTarget: 目标,
    HealAmount: 生命,
    HealManaAmount: 魔法,
    ItemHeal: true,
    HealEffect: false,
    UseDefaultHealEffect: false,
    HealEffectPath: undefined,
    ManaEffect: false,
    UseDefaultManaEffect: false,
    ManaEffectPath: undefined,
  });
}

export {};

File: src/系统/03．技能系统/00．技能模板+函数/01．技能函数/20．物品辅助/14．数值伤害与治疗.lua
local ____lualib = require("lualib_bundle")
local __TS__ArraySplice = ____lualib.__TS__ArraySplice
local ____exports = {}
local ____12_FF0E_7269_54C1_4E0E_5355_4F4D = require("系统.03．技能系统.00．技能模板+函数.01．技能函数.20．物品辅助.12．物品与单位")
local _____5355_4F4D_5B58_6D3B = ____12_FF0E_7269_54C1_4E0E_5355_4F4D["单位存活"]
local ____08_FF0E_6280_80FD_4F24_5BB3_7CFB_7EDF = require("系统.04．伤害系统.08．技能伤害系统")
local _____9020_6210_88C5_5907_6280_80FD_4F24_5BB3 = ____08_FF0E_6280_80FD_4F24_5BB3_7CFB_7EDF["造成装备技能伤害"]
local jass = require("jass.common")
local japi = require("jass.japi")
local ____require_result_0 = require("lib.扩展函数.物品相关函数.物品判断函数")
local UnitHasItemOfTypeBJ = ____require_result_0.UnitHasItemOfTypeBJ
local GetItemOfTypeFromUnitBJ = ____require_result_0.GetItemOfTypeFromUnitBJ
local ____require_result_1 = require("lib.扩展函数.自定义扩展函数.01．选取中心范围")
local getUnitsInRange = ____require_result_1.getUnitsInRange
local getEnemyUnitsInRange = ____require_result_1.getEnemyUnitsInRange
local ____require_result_2 = require("lib.扩展函数.Star扩展函数.00．SGSS")
local SGSS_SetState = ____require_result_2.SGSS_SetState
local ____require_result_3 = require("lib.扩展函数.YDWE函数.09．YDUserData安全版")
local YDUserDataGetSafe = ____require_result_3.YDUserDataGetSafe
local YDUserDataSetSafe = ____require_result_3.YDUserDataSetSafe
local ____require_result_4 = require("系统.04．伤害系统.02．治疗系统.01．核心功能")
local doHeal = ____require_result_4.doHeal
local ____require_result_5 = require("系统.03．技能系统.00．技能模板+函数.02．通用函数.01．控制与Buff")
local SFB_setBuff = ____require_result_5.SFB_setBuff
local SFB_setSlow = ____require_result_5.SFB_setSlow
local ____require_result_6 = require("系统.03．技能系统.00．技能模板+函数.02．通用函数.01．控制与Buff")
local _____6E05_9664_5355_4F4D_8D1F_9762Buff = ____require_result_6["清除单位负面Buff"]
local ____require_result_7 = require("系统.03．技能系统.00．技能模板+函数.01．技能函数.02．冲锋·击退.击退系统")
local _____5F00_59CB_51FB_9000 = ____require_result_7["开始击退"]
local ____require_result_8 = require("系统.00．核心系统.05．中心计时器")
local addPeriodicCallback = ____require_result_8.addPeriodicCallback
local getServerTime = ____require_result_8.getServerTime
local ____require_result_9 = require("lib.扩展函数.自定义扩展函数.05．单位相关安全包装")
local _____521B_5EFA_5355_4F4D_5E76_767B_8BB0_6392_6CC4_5B89_5168 = ____require_result_9["创建单位并登记排泄安全"]
local GetItemTypeId = jass.GetItemTypeId
local GetHandleId = jass.GetHandleId
local GetOwningPlayer = jass.GetOwningPlayer
local GetPlayerId = jass.GetPlayerId
local GetUnitX = jass.GetUnitX
local GetUnitY = jass.GetUnitY
local GetUnitState = jass.GetUnitState
local SetUnitState = jass.SetUnitState
local IsUnitType = jass.IsUnitType
local IsUnitAlly = jass.IsUnitAlly
local GetHeroStr = jass.GetHeroStr
local GetHeroAgi = jass.GetHeroAgi
local GetHeroInt = jass.GetHeroInt
local AddHeroXP = jass.AddHeroXP
local ModifyHeroStat = jass.ModifyHeroStat
local AddSpecialEffect = jass.AddSpecialEffect
local AddSpecialEffectTarget = jass.AddSpecialEffectTarget
local DestroyEffect = jass.DestroyEffect
local IsPointBlighted = jass.IsPointBlighted
local SetItemCharges = jass.SetItemCharges
local GetItemCharges = jass.GetItemCharges
local UnitApplyTimedLife = jass.UnitApplyTimedLife
local SetUnitScale = jass.SetUnitScale
local SetUnitInvulnerable = jass.SetUnitInvulnerable
local SetUnitFacing = jass.SetUnitFacing
local IssueTargetOrder = jass.IssueTargetOrder
local CreateGroup = jass.CreateGroup
local GroupEnumUnitsInRange = jass.GroupEnumUnitsInRange
local FirstOfGroup = jass.FirstOfGroup
local GroupRemoveUnit = jass.GroupRemoveUnit
local DestroyGroup = jass.DestroyGroup
local GetUnitFlyHeight = jass.GetUnitFlyHeight
local ConvertUnitState = jass.ConvertUnitState
local SquareRoot = jass.SquareRoot
local Atan2 = jass.Atan2
local Cos = jass.Cos
local Sin = jass.Sin
local bj_RADTODEG = jass.bj_RADTODEG
local bj_DEGTORAD = jass.bj_DEGTORAD
local UNIT_TYPE_HERO = jass.UNIT_TYPE_HERO
local UNIT_TYPE_DEAD = jass.UNIT_TYPE_DEAD
local UNIT_TYPE_MECHANICAL = jass.UNIT_TYPE_MECHANICAL
local UNIT_TYPE_ANCIENT = jass.UNIT_TYPE_ANCIENT
local UNIT_STATE_LIFE = jass.UNIT_STATE_LIFE
local UNIT_STATE_MANA = jass.UNIT_STATE_MANA
local UNIT_STATE_MAX_LIFE = jass.UNIT_STATE_MAX_LIFE
local UNIT_STATE_MAX_MANA = jass.UNIT_STATE_MAX_MANA
local ATTACK_TYPE_NORMAL = jass.ATTACK_TYPE_NORMAL
local DAMAGE_TYPE_NORMAL = jass.DAMAGE_TYPE_NORMAL
local DAMAGE_TYPE_ENHANCED = jass.DAMAGE_TYPE_ENHANCED
local DAMAGE_TYPE_FIRE = jass.DAMAGE_TYPE_FIRE
local DAMAGE_TYPE_SHADOW_STRIKE = jass.DAMAGE_TYPE_SHADOW_STRIKE
local DAMAGE_TYPE_MIND = jass.DAMAGE_TYPE_MIND
local WEAPON_TYPE_WHOKNOWS = jass.WEAPON_TYPE_WHOKNOWS
local bj_HEROSTAT_INT = jass.bj_HEROSTAT_INT
local bj_MODIFYMETHOD_ADD = jass.bj_MODIFYMETHOD_ADD
local GetUnitStateJapi = japi.GetUnitState
local DzSetUnitModel = japi.DzSetUnitModel
local stringToFourCCSafe = require("lib.扩展函数.封装函数.01．通用工具.01．FourCC转换安全版").stringToFourCCSafe
local _____706B_628A_5355_4F4D_7C7B_578BID = stringToFourCCSafe("e0FT")
local _____9650_65F6_751F_547DBuffID = stringToFourCCSafe("BHwe")
local _____5F85_9500_6BC1_7279_6548_5217_8868 = {}
local _____5DF2_6CE8_518C_7279_6548_9500_6BC1_9A71_52A8 = false
local function _____5904_7406_5F85_9500_6BC1_7279_6548()
    local _____5F53_524D_65F6_95F4 = getServerTime()
    do
        local i = #_____5F85_9500_6BC1_7279_6548_5217_8868 - 1
        while i >= 0 do
            do
                local _____8BB0_5F55 = _____5F85_9500_6BC1_7279_6548_5217_8868[i + 1]
                if _____5F53_524D_65F6_95F4 < _____8BB0_5F55["到期时间"] then
                    goto __continue4
                end
                DestroyEffect(_____8BB0_5F55["句柄"])
                __TS__ArraySplice(_____5F85_9500_6BC1_7279_6548_5217_8868, i, 1)
            end
            ::__continue4::
            i = i - 1
        end
    end
end
local function _____5B89_6392_7279_6548_9500_6BC1(effect, _____6301_7EED_79D2)
    if _____6301_7EED_79D2 == nil then
        _____6301_7EED_79D2 = 1
    end
    if effect == nil or effect == 0 then
        return
    end
    if not _____5DF2_6CE8_518C_7279_6548_9500_6BC1_9A71_52A8 then
        _____5DF2_6CE8_518C_7279_6548_9500_6BC1_9A71_52A8 = true
        addPeriodicCallback(100, _____5904_7406_5F85_9500_6BC1_7279_6548)
    end
    _____5F85_9500_6BC1_7279_6548_5217_8868[#_____5F85_9500_6BC1_7279_6548_5217_8868 + 1] = {
        ["句柄"] = effect,
        ["到期时间"] = getServerTime() + _____6301_7EED_79D2 * 1000
    }
end
____exports["取当前生命"] = function(_____5355_4F4D)
    return GetUnitState(_____5355_4F4D, UNIT_STATE_LIFE)
end
____exports["取当前魔法"] = function(_____5355_4F4D)
    return GetUnitState(_____5355_4F4D, UNIT_STATE_MANA)
end
____exports["取最大生命"] = function(_____5355_4F4D)
    return GetUnitStateJapi(_____5355_4F4D, UNIT_STATE_MAX_LIFE)
end
____exports["取最大魔法"] = function(_____5355_4F4D)
    return GetUnitStateJapi(_____5355_4F4D, UNIT_STATE_MAX_MANA)
end
____exports["取单位攻击"] = function(_____5355_4F4D)
    return GetUnitStateJapi(
        _____5355_4F4D,
        ConvertUnitState(21)
    )
end
____exports["设置生命"] = function(_____5355_4F4D, _____6570_503C)
    SetUnitState(_____5355_4F4D, UNIT_STATE_LIFE, _____6570_503C)
end
____exports["设置魔法"] = function(_____5355_4F4D, _____6570_503C)
    SetUnitState(_____5355_4F4D, UNIT_STATE_MANA, _____6570_503C)
end
____exports["调整生命"] = function(_____5355_4F4D, _____6570_503C)
    SetUnitState(
        _____5355_4F4D,
        UNIT_STATE_LIFE,
        GetUnitState(_____5355_4F4D, UNIT_STATE_LIFE) + _____6570_503C
    )
end
____exports["调整魔法"] = function(_____5355_4F4D, _____6570_503C)
    SetUnitState(
        _____5355_4F4D,
        UNIT_STATE_MANA,
        GetUnitState(_____5355_4F4D, UNIT_STATE_MANA) + _____6570_503C
    )
end
____exports["造成强化伤害"] = function(_____6765_6E90, _____76EE_6807, _____4F24_5BB3)
    if not _____5355_4F4D_5B58_6D3B(_____6765_6E90) or not _____5355_4F4D_5B58_6D3B(_____76EE_6807) or not (_____4F24_5BB3 > 0) then
        return
    end
    _____9020_6210_88C5_5907_6280_80FD_4F24_5BB3({
        ["来源"] = _____6765_6E90,
        ["目标"] = _____76EE_6807,
        ["伤害"] = _____4F24_5BB3,
        ["伤害类型"] = DAMAGE_TYPE_ENHANCED,
        ranged = false,
        attackType = ATTACK_TYPE_NORMAL,
        weaponType = WEAPON_TYPE_WHOKNOWS,
        ["伤害形态"] = "单体"
    })
end
____exports["造成火焰伤害"] = function(_____6765_6E90, _____76EE_6807, _____4F24_5BB3)
    if not _____5355_4F4D_5B58_6D3B(_____6765_6E90) or not _____5355_4F4D_5B58_6D3B(_____76EE_6807) or not (_____4F24_5BB3 > 0) then
        return
    end
    _____9020_6210_88C5_5907_6280_80FD_4F24_5BB3({
        ["来源"] = _____6765_6E90,
        ["目标"] = _____76EE_6807,
        ["伤害"] = _____4F24_5BB3,
        ["伤害类型"] = DAMAGE_TYPE_FIRE,
        ranged = true,
        attackType = ATTACK_TYPE_NORMAL,
        weaponType = WEAPON_TYPE_WHOKNOWS,
        ["伤害形态"] = "单体"
    })
end
____exports["造成暗影伤害"] = function(_____6765_6E90, _____76EE_6807, _____4F24_5BB3)
    if not _____5355_4F4D_5B58_6D3B(_____6765_6E90) or not _____5355_4F4D_5B58_6D3B(_____76EE_6807) or not (_____4F24_5BB3 > 0) then
        return
    end
    _____9020_6210_88C5_5907_6280_80FD_4F24_5BB3({
        ["来源"] = _____6765_6E90,
        ["目标"] = _____76EE_6807,
        ["伤害"] = _____4F24_5BB3,
        ["伤害类型"] = DAMAGE_TYPE_SHADOW_STRIKE,
        ranged = true,
        attackType = ATTACK_TYPE_NORMAL,
        weaponType = WEAPON_TYPE_WHOKNOWS,
        ["伤害形态"] = "单体"
    })
end
____exports["造成普通伤害"] = function(_____6765_6E90, _____76EE_6807, _____4F24_5BB3)
    if not _____5355_4F4D_5B58_6D3B(_____6765_6E90) or not _____5355_4F4D_5B58_6D3B(_____76EE_6807) or not (_____4F24_5BB3 > 0) then
        return
    end
    _____9020_6210_88C5_5907_6280_80FD_4F24_5BB3({
        ["来源"] = _____6765_6E90,
        ["目标"] = _____76EE_6807,
        ["伤害"] = _____4F24_5BB3,
        ["伤害类型"] = DAMAGE_TYPE_NORMAL,
        ranged = false,
        attackType = ATTACK_TYPE_NORMAL,
        weaponType = WEAPON_TYPE_WHOKNOWS,
        ["伤害形态"] = "单体"
    })
end
____exports["造成精神自伤"] = function(_____5355_4F4D, _____4F24_5BB3)
    if not _____5355_4F4D_5B58_6D3B(_____5355_4F4D) or not (_____4F24_5BB3 > 0) then
        return
    end
    _____9020_6210_88C5_5907_6280_80FD_4F24_5BB3({
        ["来源"] = _____5355_4F4D,
        ["目标"] = _____5355_4F4D,
        ["伤害"] = _____4F24_5BB3,
        ["伤害类型"] = DAMAGE_TYPE_MIND,
        ranged = false,
        attackType = ATTACK_TYPE_NORMAL,
        weaponType = WEAPON_TYPE_WHOKNOWS,
        ["伤害形态"] = "单体"
    })
end
____exports["执行治疗"] = function(_____6765_6E90, _____76EE_6807, _____751F_547D, _____9B54_6CD5)
    if _____9B54_6CD5 == nil then
        _____9B54_6CD5 = 0
    end
    if _____76EE_6807 == nil or _____76EE_6807 == 0 then
        return
    end
    doHeal({
        HealSource = _____6765_6E90,
        HealTarget = _____76EE_6807,
        HealAmount = _____751F_547D,
        HealManaAmount = _____9B54_6CD5,
        ItemHeal = true,
        HealEffect = _____751F_547D > 0,
        UseDefaultHealEffect = _____751F_547D > 0,
        HealEffectPath = nil,
        ManaEffect = _____9B54_6CD5 > 0,
        UseDefaultManaEffect = _____9B54_6CD5 > 0,
        ManaEffectPath = nil
    })
end
return ____exports
